### PR TITLE
protecting against empty strings in events

### LINKED ISSUE #8: TypeError: Cannot call method 'split' of undefined
I can't seem to get statsd-alerting-backend to work.

1. Start StatsD with statsd-alerting-backend using Slack dispatcher.
2. echo "test.localcounter:100|c" | nc -u -w 1 127.0.0.1 8125

Statsd log:

```
1 Sep 16:02:29 - DEBUG: Loading backend: statsd-alerting-backend

/Users/chenharel/Sparktale/statsd/node_modules/statsd-alerting-backend/lib/alert_distributor.js:101
      ref2 = data.split('|'), metric = ref2[0], type = ref2[1];
                  ^
TypeError: Cannot call method 'split' of undefined
    at AlertDistributor.module.exports.AlertDistributor.parsePacket (/Users/chenharel/Sparktale/statsd/node_modules/statsd-alerting-backend/lib/alert_distributor.js:101:19)
    at AlertDistributor.module.exports.AlertDistributor.onPacket (/Users/chenharel/Sparktale/statsd/node_modules/statsd-alerting-backend/lib/alert_distributor.js:128:16)
    at EventEmitter.<anonymous> (/Users/chenharel/Sparktale/statsd/node_modules/statsd-alerting-backend/lib/alert_distributor.js:2:57)
    at EventEmitter.emit (events.js:98:17)
    at Socket.<anonymous> (/Users/chenharel/Sparktale/statsd/node_modules/statsd/stats.js:172:21)
    at Socket.EventEmitter.emit (events.js:98:17)
    at UDP.onMessage (dgram.js:440:8)
```

### PR DESCRIPTION
**Pull request by chook**
Fixed #8

### CODE REVIEW

File: src/alert_distributor.coffee [hunk 1]
```diff
@@ -48,6 +48,7 @@ module.exports = class AlertDistributor
   # Parse out event data from StatsD packet
   parsePacket: (packet) ->
     for event in packet.toString().split('\n') or []
+      continue if !event
```
Comment (joshgummersall) [inline src/alert_distributor.coffee]: I'd write this as `continue unless event` as it reads a bit more naturally.

Comment (chook) [inline src/alert_distributor.coffee]: thanks! well i hacked the js output to make it work and then tried my best to write coffee :8ball:

Comment (joshgummersall) [inline src/alert_distributor.coffee]: No sweat - I appreciate the contribution!

File: src/alert_distributor.coffee [hunk 2]
```diff
@@ -103,7 +104,7 @@ module.exports = class AlertDistributor
   extractMatchedMetrics: (metricsEvent, metrics) ->
     {type, name, key} = metricsEvent
     subMetrics = metrics?[type]
-    return unless subMetrics
+    return [] unless subMetrics
```
Comment (joshgummersall) [inline src/alert_distributor.coffee]: Good find!

**Comment by joshgummersall** [comment]
Great work @chook thanks again for contributing!

File: src/alert_distributor.coffee (hunk 2 above)
Comment (chook) [inline src/alert_distributor.coffee]: :smiley:

**Pull request merged**